### PR TITLE
Do not use empty region

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**

Changes proposed in this pull request:
- if user provides an empty region in provisioning parameters, KEB should do the same as the region is nil

### CODE REVIEW

File: components/kyma-environment-broker/internal/process/input/input.go [hunk 1]
```diff
@@ -428,7 +428,9 @@ func (r *RuntimeInput) applyProvisioningParametersForProvisionRuntime() error {
 	updateInt(&r.provisionRuntimeInput.ClusterConfig.GardenerConfig.AutoScalerMax, params.AutoScalerMax)
 	updateInt(r.provisionRuntimeInput.ClusterConfig.GardenerConfig.VolumeSizeGb, params.VolumeSizeGb)
 	updateString(&r.provisionRuntimeInput.ClusterConfig.GardenerConfig.Name, r.shootName)
-	updateString(&r.provisionRuntimeInput.ClusterConfig.GardenerConfig.Region, params.Region)
+	if params.Region != nil && *params.Region != "" {
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/process/input/input.go:431]: I'm thinking about having this "if" statement inside the updateString function, but not sure if it breaks something

File: components/kyma-environment-broker/cmd/broker/provisioning_test.go [hunk 2]
```diff
@@ -42,6 +42,38 @@ func TestProvisioning_HappyPath(t *testing.T) {
...
+							  "url": "https://sm.url",
+							  "credentials": {}
+					    },
+						"globalaccount_id": "g-account-id",
+						"subaccount_id": "sub-id",
+						"user_id": "john.smith@email.com"
+					},
+					"parameters": {
+						"name": "testing-cluster",
+                        "region":""
```
Comment (wozniakjan) [inline components/kyma-environment-broker/cmd/broker/provisioning_test.go:67]: nit: there is a mix of tabs and spaces here

File: components/kyma-environment-broker/internal/process/input/input.go (hunk 1 above)
Comment (wozniakjan) [reply components/kyma-environment-broker/internal/process/input/input.go:431]: yeah, I agree that for some arguments there could be a difference between providing empty value or not providing value at all, probably safer to keep it separate.

File: components/kyma-environment-broker/internal/provider/azure_provider.go [hunk 3]
```diff
@@ -144,8 +146,11 @@ func (p *AzureTrialInput) ApplyParameters(input *gqlschema.ClusterConfigInput, p
 		}
 	}
 
-	if params.Region != nil {
+	if params.Region != nil && *params.Region != "" {
+		fmt.Println(*params.Region)
 		updateString(&input.GardenerConfig.Region, toAzureSpecific[*params.Region])
+		fmt.Println(input.GardenerConfig.Region)
+		fmt.Println(toAzureSpecific[*params.Region])
```
Comment (wozniakjan) [inline components/kyma-environment-broker/internal/provider/azure_provider.go:153]: nit: are these `fmt.Println` statements intended to stay?

**Review by wozniakjan — COMMENTED** [review]
implementation looks solid, just take a look at the nits provided below please

File: components/kyma-environment-broker/internal/provider/azure_provider.go (hunk 3 above)
Comment (piotrmiskiewicz) [reply components/kyma-environment-broker/internal/provider/azure_provider.go]: removed

File: components/kyma-environment-broker/cmd/broker/provisioning_test.go (hunk 2 above)
Comment (piotrmiskiewicz) [reply components/kyma-environment-broker/cmd/broker/provisioning_test.go]: fixed

**Review by wozniakjan — APPROVED** [review]

**Pull request merged**